.github/workflows/pythonpackage.yml

```diff
@@ -55,31 +55,26 @@ jobs:
       run: |
         poetry install --extras docs 
 
-    - name: Run pytest
-      shell: bash
-      run: |
-        poetry run python -m pytest --cov=fgpyo --cov-report=xml --cov-branch 
-
-    - name: Run isort
-      shell: bash
-      run: |
-        poetry run isort --force-single-line-imports --profile black fgpyo
-
     - name: Style checking
       shell: bash
       run: |
-        poetry run black --line-length 99 --check fgpyo
+        poetry run ruff format --check fgpyo
 
     - name: Run lint
       shell: bash
       run: |
-        poetry run flake8 --config=ci/flake8.cfg fgpyo
+        poetry run ruff check fgpyo
 
     - name: Run mypy
       shell: bash
       run: |
         poetry run mypy -p fgpyo --config=ci/mypy.ini
 
+    - name: Run pytest
+      shell: bash
+      run: |
+        poetry run python -m pytest --cov=fgpyo --cov-report=xml --cov-branch 
+
```

Comment:
I find formatting, linting, type-checking, and unit testing to be a better order of operations. 

1) it's loosely ordered by how long I expect each to take (put quicker steps first to fail fast, especially on GHA), 

2) it's the order in which I think about fixing issues with the code. It's easier to debug a test when you're confident in the type-checking, and type checks are easier to fix when you're not also wrangling with lints or style issues